### PR TITLE
Music GUI improvements

### PR DESCRIPTION
**Pull request by nielsmh**
Various small improvements to music GUI:
- Button in Music Programme window to change music set during gameplay
- Less annoying behavior when an empty music set is loaded
- Fix some clipping issues with the volume sliders

### CODE REVIEW

File: src/music_gui.cpp [hunk 1]
```diff
@@ -799,6 +804,5 @@ static WindowDesc _music_window_desc(
 
 void ShowMusicWindow()
 {
-	if (BaseMusic::GetUsedSet()->num_available == 0) ShowErrorMessage(STR_ERROR_NO_SONGS, INVALID_STRING_ID, WL_WARNING);
```
Comment (frosch123) [inline src/music_gui.cpp:779]: STR_ERROR_NO_SONGS is also unused now, please also remove it in the "Cleanup" commit. (btw. "Cleanup" fits better than "Update" here)

File: src/music_gui.cpp [hunk 2]
```diff
@@ -612,15 +612,13 @@ struct MusicWindow : public Window {
 			}
 
 			case WID_M_MUSIC_VOL: case WID_M_EFFECT_VOL: {
-				DrawFrameRect(r.left, r.top + 2, r.right, r.bottom - 2, COLOUR_GREY, FR_LOWERED);
+				int sw = slider_width * (1 << (ZOOM_LVL_OUT_4X - _gui_zoom));
```
Comment (frosch123) [inline src/music_gui.cpp]: Please use ScaleGUITrad() here instead of the magic shifting.

**Comment by frosch123** [comment]
The UpdateDisabledButtons part could already be done in "Change: Disable music control buttons when there is no music". It does not belong to "Feature #6090: Change music set during gameplay".

File: src/music_gui.cpp [hunk 3]
```diff
@@ -312,12 +314,30 @@ struct MusicTrackSelectionWindow : public Window {
 		this->LowerWidget(WID_MTS_ALL + _settings_client.music.playlist);
 	}
 
+	static DropDownList *BuildMusicSetsDropdown(int &selected_index)
+	{
+		DropDownList *musicsets = new DropDownList();
```
Comment (frosch123) [inline src/music_gui.cpp]: Please ensure that "selected_index" gets initialised, without relying on BaseMusic::GetUsedSet not returning NULL or something

File: src/music_gui.cpp [hunk 4]
```diff
@@ -470,12 +497,37 @@ struct MusicTrackSelectionWindow : public Window {
 				break;
 		}
 	}
+
+	virtual void OnDropdownClose(Point pt, int widget, int index, bool instant_close)
+	{
```
Comment (frosch123) [inline src/music_gui.cpp]: I think you wanted to implement OnDropdownSelect.
The default OnDropdownClose method should be fine.

File: src/music_gui.cpp (hunk 2 above)
Comment (nielsmh) [reply src/music_gui.cpp]: Thanks for the heads up. It did smell fishy but I had no idea where to look for the proper way.

**Comment by frosch123** [comment]
The first three commits look great now :)

About changing music sets: There is also the game options window, which allows changing music sets.
But currently they only allow changing the music in the main menu, while the dropdown is read-only in-game.
I also noticed some code-duplication between your new BuildMusicSetsDropdown/OnDropdownSelect and the corresponding functions BuiltSetDropDownList/GameOptionsWindow::SetMediaSet.

Is it possible to unify the behaviour of these windows?
* GameOptionsWindow should also be able to change music in-game.
* Building the dropdown and switching the active set should use the same code/function.

**Comment by nielsmh** [comment]
The best approach is probably to just never disable the dropdown in Game Options, and move most of the stopping/restarting/playlist resetting logic into a separate function, that can then be called from the respective `OnDropdownSelect`. Maybe move `BuiltSetDropDownList` from `settings_gui.cpp` to `settings_gui.h` and make it more flexible in disabled-ness handling.

File: src/music_gui.cpp [hunk 5]
```diff
@@ -471,12 +514,25 @@ struct MusicTrackSelectionWindow : public Window {
 				break;
 		}
 	}
+
+	virtual void OnDropdownSelect(int widget, int index)
+	{
+		switch (widget) {
+		case WID_MTS_MUSICSET:
```
Comment (frosch123) [inline src/music_gui.cpp]: We add an additional indent level after switch.

**Review by frosch123 — COMMENTED** [review]
I only found one indentation style issue :)

Then it's ready.

**Review by frosch123 — APPROVED** [review]

**Pull request merged**